lib/rtc_peer.js

```diff
@@ -0,0 +1,195 @@
+var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
+    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
+    return new (P || (P = Promise))(function (resolve, reject) {
+        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
+        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
+        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
+        step((generator = generator.apply(thisArg, _arguments || [])).next());
+    });
+};
+import { EventEmitter } from 'events';
+const rtcConnFailedErr = new Error('rtc connection failed');
+export class RTCPeer extends EventEmitter {
+    constructor(config) {
+        super();
+        this.makingOffer = false;
+        this.candidates = [];
+        this.logDebug = config.logDebug;
+        this.logErr = config.logErr;
```

Comment:
Would it make sense to accept some kind of logger object that provides these methods? Mostly to avoid adding a new variable every time we want a new log level.